### PR TITLE
Feature/storage/set file tier

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

**Comment by jsquire** [comment]
Hi @seanmcc-msft.  There hasn't been recent engagement on this PR.  Would you please be so kind as to let us know if this is still an active work stream or if the PR should be closed out?

**Comment by seanmcc-msft** [comment]
Please do not close this PR, we are waiting on the service.

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareServiceClient.cs [hunk 1]
```diff
@@ -645,8 +646,100 @@ internal ShareServiceClient(Uri serviceUri, HttpPipelinePolicy authentication, S
...
+        /// A <see cref="Response{ShareClient}"/> referencing the newly
+        /// created share.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<ShareClient> CreateShare(
+            string shareName,
+            ShareCreateOptions options,
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/src/ShareServiceClient.cs:677]: if `options` is optional, should it be `ShareCreateOptions options = default` ?

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/ShareServiceClient.cs:677]: I think we need this to prevent ambiguity with the other overloads.

**Comment by jsquire** [comment]
Thanks, @seanmcc-msft.  Untagged, and I'll leave this be.

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs [hunk 2]
```diff
@@ -386,6 +399,25 @@ public partial class ShareAccessPolicy
...
+        public bool Equals(Azure.Storage.Files.Shares.Models.ShareAccessTier other) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object obj) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        public static bool operator ==(Azure.Storage.Files.Shares.Models.ShareAccessTier left, Azure.Storage.Files.Shares.Models.ShareAccessTier right) { throw null; }
+        public static implicit operator Azure.Storage.Files.Shares.Models.ShareAccessTier (string value) { throw null; }
+        public static bool operator !=(Azure.Storage.Files.Shares.Models.ShareAccessTier left, Azure.Storage.Files.Shares.Models.ShareAccessTier right) { throw null; }
+        public override string ToString() { throw null; }
+    }
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs:420]: should we create enum for that or exposing generated type is ok ?

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs:420]: This is how we handled AccessTier in Blobs: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs#L167

**Review by kasobol-msft — APPROVED** [review]

**Pull request merged**